server_installation/topics/operator/keycloak-cr.adoc

```diff
@@ -0,0 +1,157 @@
+* credentials-<CR-Name> - Admin user credentials to log into the {project_name} installation
+* keycloak StatefulSet with HA support
+* keycloak-postgresql - Responsible for starting a PostgreSQL installation
+* keycloak-discovery Service - Used for JDBC_PING discovery
+* keycloak Service - Used for connecting to {project_name} through HTTPS (HTTP is not allowed)
+* keycloak-postgresql - A service for connecting both an internal and external, if used, database instance
+ifeval::[{project_product}==true]
+* keycloak Route - Used for accessing {project_name} on OpenShift
+endif::[]
+ifeval::[{project_community}==true]
+* keycloak Ingress - Used for accessing {project_name} on Kubernetes
+endif::[]
+
+The following graphic illustrates how these components interact.
+
+[NOTE]
+This graphic is a place holder for one to be developed by our graphic artist.
+
+image:{project_images}/operator-components.png[]
+
+MyKeycloak name, which is used for a Keycloak CR, appears only in the credential-[CR Name]. Since the Operator can be installed only on one {project_name} cluster in a namespace, it is always called keycloak.
+
+You can use the Operator to manage {project_name} clusters by creating a Keycloak CR on
+ifeval::[{project_community}==true]
+Kubernetes or
+endif::[]
+Openshift. You define the properties of the Keycloak CR in a YAML file. You can update the YAML file and changes appear in the {project_name} Admin Console, however changes to the Admin Console do not update the CR.
+
+.Example YAML file for a Keycloak CR
+```yaml
+apiVersion: keycloak.org/v1alpha1
+kind: Keycloak
+metadata:
+  name: example-keycloak
+  labels:
+    app: sso
+spec:
+  instances: 1
+  externalAccess:
+    enabled: True
+```
+
+The `Keycloak` YAML file contains three properties. Two properties are listed above and the external database is covered in a seperate section in this chapter:
+
+* `instances` - controls the number of instances running in high avaialability mode
+* `externalAccess` - if the `enabled` is `True`, the Operator will create a route for OpenShift
+ifeval::[{project_community}==true]
+ or an Ingress for Kubernetes
+endif::[]
+ for the {project_name} cluster.
+
+.Prerequisite
+
+You have a YAML file for this CR. See link:https://github.com/keycloak/keycloak-operator/tree/master/deploy/examples[Sample YAML files].
+
+ifeval::[{project_community}==true]
+.Procedure for Kubernetes
+
+. Use the `kubectl` command on the YAML file that you created.
++
+oc create -f <filename>.yaml -n <namespace>
++
+This command will create an ingress on Kubernetes.
+
+. Use the `kubectl` command to find the ingress created by the `oc` command.
++
+kubectl get ingress -n <cr_name>
+
+. Locate the password by using this command
++
+kubectl get secret credentials-<CR-Name> -o go-template='{{range $k,$v := .data}}{{printf "%s: " $k}}{{if not $v}}{{$v}}{{else}}{{$v | base64decode}}{{end}}{{"\n"}}{{end}}'
```

Comment:
This needs to be a code snippet. The same applies for other bits in this paragraph (I won't be commenting them one by one).

Reply:
Yes, that makes sense.  I can make that change globally.

Reply:
Thanks

Reply:
I added the snippet code, @slaskawi

Reply:
Thanks